docs/porting/custom-target-howto.md

```diff
@@ -0,0 +1,117 @@
+
+The first step in adding a custom Mbed target for your custom board is to determine which MCU your custom board uses. Let's say your board uses, for the sake of example, an NRF52840 MCU.
+
+Now that you know your MCU, you need to locate the MCU CMake target within Mbed OS. For the NRF52840, the CMake target is specified in [the CMake file for the NRF52840](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/CMakeLists.txt). All available MCU CMake targets are available within [Mbed OS's `targets/`](https://github.com/ARMmbed/mbed-os/tree/master/targets) directory. If you can't find your desired MCU, [we'd welcome a PR adding it](https://os.mbed.com/docs/mbed-os/latest/contributing/index.html)! Once you've found the MCU's CMake target name, please take note of the name as we'll need to refer to it again soon. 
+
+Let's get back to your application. We need to create your new target and tell both CMake and Mbed about it. We'll start with telling Mbed.
+
+### Telling Mbed about your custom board
+
+To tell Mbed about your target, create a new file called `custom_targets.json` in your application's directory.
+
+```
+mbed-os-example-blinky
+├── CMakeLists.txt
+├── CONTRIBUTING.md
+├── custom_targets.json     <--- The new file!
+├── LICENSE
+├── main.cpp
+├── mbed-os.lib
+├── README.md
+└── resources
+    └── official_armmbed_example_badge.png
+```
+
+Inside this new file, you can use Mbed's configuration system to specify extra information about your board. An example is below. [Our target documentation](https://os.mbed.com/docs/mbed-os/latest/program-setup/adding-and-configuring-targets.html) provides further information on what configuration options are available for targets.
+
+```
+{
+    "IMAGINARY": {
+        "inherits": [
+            "NRF52840"
+        ]
+    }
+}
+```
+
+#### Making the CMake target
+
+Next, we need to tell CMake about your custom target.
+
+The most common thing to add for custom targets is a custom pin configuration. As such, we'll assume you'll be wanting to customize your custom board's pin configuration too. To do this, you'll need to make a few files: at least a `PinNames.h` and possibly also a `PeripheralPins.h` or a `PeripheralNames.h`. Please refer to [the Mbed OS porting guide](https://os.mbed.com/docs/mbed-os/latest/porting/porting-custom-boards.html) on how to configure the pins for your board and which files exactly you'd need to create.
+
+```
+mbed-os-example-blinky
+├── CMakeLists.txt
+├── CONTRIBUTING.md
+├── custom_targets.json
+├── IMAGINARY              <--- The new folder with new files in it!
```

Comment:
Currently, we follow `TARGET_<BOARD_NAME>` naming convention for board/target related files. Can we follow the same convention in the example as well?

Reply:
The targets aren't named TARGET_BLAH. The target name is actually just BLAH. For folder names, given the limitation of the old tools, we did name things TARGET_BLAH. I'll do that here for old time's sake.